vendor/github.com/distribution/reference/normalize.go

```diff
@@ -123,20 +123,51 @@ func ParseDockerRef(ref string) (Named, error) {
 // splitDockerDomain splits a repository name to domain and remote-name.
 // If no valid domain is found, the default domain is used. Repository name
 // needs to be already validated before.
-func splitDockerDomain(name string) (domain, remainder string) {
-	i := strings.IndexRune(name, '/')
-	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != localhost && strings.ToLower(name[:i]) == name[:i]) {
-		domain, remainder = defaultDomain, name
-	} else {
-		domain, remainder = name[:i], name[i+1:]
-	}
-	if domain == legacyDefaultDomain {
-		domain = defaultDomain
-	}
-	if domain == defaultDomain && !strings.ContainsRune(remainder, '/') {
-		remainder = officialRepoPrefix + remainder
-	}
-	return
+func splitDockerDomain(name string) (domain, remoteName string) {
```

Comment:
do we have pointers to the upstream PRs this is pulling in? we use this 

```
pkg/apis/core/v1/defaults.go
  74,26: 		_, tag, _, _ := parsers.ParseImageName(obj.Image.Reference)
  85,26: 		_, tag, _, _ := parsers.ParseImageName(obj.Image)

pkg/kubelet/images/image_manager.go
  208,32: 	_, tag, digest, err := parsers.ParseImageName(image)

pkg/kubelet/kuberuntime/kuberuntime_image.go
  37,34: 	repoToPull, _, _, err := parsers.ParseImageName(img)
```

trying to understand the changes to this function and if it is erroring or relaxing in new ways

Reply:
@liggitt not sure I understand the question - what do you mean by "upstream PRs this is pulling in" ? 

https://github.com/distribution/reference/commit/89ee7ec63e17bd74e6766ecac7d8ff2bd4b8dd8a

Reply:
https://github.com/distribution/reference/compare/v0.5.0...v0.6.0

trying to understand if this is a refactor without a behavior change, or if it is erroring in new ways, or stopping erroring where it used to

Reply:
maybe @thaJeztah could help clarify (as the author of https://github.com/distribution/reference/pull/7)

Reply:
got a response at https://github.com/distribution/reference/pull/7#issuecomment-2348512110, sounds like this was just a refactor, not a behavior change 👍